numpy/core/src/multiarray/refcount.c

```diff
@@ -258,6 +308,10 @@ _fillobject(char *optr, PyObject *obj, PyArray_Descr *dtype)
             Py_XDECREF(arr);
         }
     }
+    if (dtype->type_num == NPY_OBJECT) {
+        Py_XINCREF(obj);
```

Comment:
Surely `obj` cannot be `NULL` here?

Reply:
@seberg do you want to change this?

Reply:
@mattip, I don't mind changing this. I am just not sure how much we implicitly support this, since NULL is also always checked e.g. in the ufunc loops that you worked on with the exception chaining.
But, I am pretty ambivalent about it. If you/anyone thinks we should just move away from it, lets do that. I do not really see the point in it.